tests/common/include/tests/common/material_interaction.inl

```diff
@@ -0,0 +1,450 @@
+TEST(material_interaction, telescope_geometry_scattering_angle) {
+    vecmem::host_memory_resource host_mr;
+
+    using ln_stepper_t = line_stepper<transform3>;
+
+    typename ln_stepper_t::free_track_parameters_type default_trk{
+        {0, 0, 0}, 0, {1, 0, 0}, -1};
+
+    // Build from given module positions
+    std::vector<scalar> positions = {0., 1000. * unit_constants::cm};
+
+    const auto mat = silicon_tml<scalar>();
+    const scalar thickness = 500 * unit_constants::cm;
+    // Use unbounded surfaces
+    constexpr bool unbounded = true;
+
+    const auto det = create_telescope_detector<unbounded>(
+        host_mr, positions, ln_stepper_t(),
+        typename ln_stepper_t::state{default_trk}, 2000. * unit_constants::mm,
+        2000. * unit_constants::mm, mat, thickness);
+
+    using navigator_t = navigator<decltype(det)>;
+    using constraints_t = constrained_step<>;
+    using policy_t = stepper_default_policy;
+    using stepper_t = line_stepper<transform3, constraints_t, policy_t>;
+    using interactor_t = pointwise_material_interactor<transform3>;
+    using simulator_t = scattering_simulator<interactor_t>;
+    using actor_chain_t =
+        actor_chain<dtuple, propagation::print_inspector, pathlimit_aborter,
+                    parameter_transporter<transform3>, interactor_t,
+                    simulator_t, parameter_resetter<transform3>>;
+    using propagator_t = propagator<stepper_t, navigator_t, actor_chain_t>;
+
+    // Propagator is built from the stepper and navigator
+    propagator_t p({}, {});
+
+    const scalar q = -1.;
+    const scalar iniP = 10 * unit_constants::GeV;
+
+    typename bound_track_parameters<transform3>::vector_type bound_vector;
+    getter::element(bound_vector, e_bound_loc0, 0) = 0.;
+    getter::element(bound_vector, e_bound_loc1, 0) = 0.;
+    getter::element(bound_vector, e_bound_phi, 0) = 0;
+    getter::element(bound_vector, e_bound_theta, 0) = M_PI_2;
+    getter::element(bound_vector, e_bound_qoverp, 0) = q / iniP;
+    getter::element(bound_vector, e_bound_time, 0) = 0.;
+    typename bound_track_parameters<transform3>::covariance_type bound_cov =
+        matrix_operator().template zero<e_bound_size, e_bound_size>();
+
+    // bound track parameter
+    const bound_track_parameters<transform3> bound_param(0, bound_vector,
+                                                         bound_cov);
+
+    int n_samples = 100000;
+    std::vector<scalar> phi_vec;
+    std::vector<scalar> theta_vec;
+
+    scalar ref_phi_var(0);
+    scalar ref_theta_var(0);
+
+    for (int i = 0; i < n_samples; i++) {
+
+        propagation::print_inspector::state print_insp_state{};
+        pathlimit_aborter::state aborter_state{};
+        parameter_transporter<transform3>::state bound_updater{};
+        interactor_t::state interactor_state{};
+        interactor_state.do_energy_loss = false;
+        simulator_t::state simulator_state(interactor_state);
```

Comment:
@niermann999 I added a simulator actor which refers the material interactor state. Is it possible to utilize the observer chain instead of having interactor state as an argument? (Sorry my understanding on observer is quite bad)

Reply:
If I understand you correctly, then yes, definetly. From memory, you should be able to do something like this: 
```using material_actor_t = composite_actor<tuple_t, material_interactor, simulator>;
using actor_chain_t = actor_chain<tuple_t, pathlimit_aborter, material_actor_t>;

pathlimit_aborter::state pathlimit_aborter_state {path_limit};
interactor_t::state interactor_state{};
interactor_state.do_energy_loss = false;
simulator_t::state simulator_state{};

actor_chain_t::state actor_states = thrust::tie(pathlimit_aborter_state, interactor_state, simulator_state); // or std::tie

actor_chain <dtuple , actor_chain_t> run_chain {};
run_chain ( actor_states , prop_state );
```
The simulator needs to provide a call operator that takes the material interactor state as its second argument. Then, the pathlimit aborter gets called first, the material interactor gets called second and the simulator gets called last, but will also be handed the material interactor state automatically on its call.

Reply:
If you provide the right overloads in the simulator, you can implement a call operator per subject actor type (e.g. different material interactor types, if any exist), without touching the simulator state type.

Reply:
However, after the pathlimit aborter the chain only terminates early, if the following actors check the propagation heartbeat. Maybe I should put this in the actor chain directly...

Reply:
Hm, OK, on second thought that might not be that easy. Let's keep it like it is for the moment